bug: CI homeboy-action fails with 404 — chicken-and-egg binary download

## Problem

The `Homeboy Audit` CI job on PR #413 (and likely all PRs) fails with:

```
curl: (22) The requested URL returned error: 404
##[error]Process completed with exit code 22.
```

The `homeboy-action@v1` GitHub Action downloads the homeboy binary from the **latest GitHub release**. When the latest release doesn't have a binary for the target platform (or the release tag doesn't exist yet), the download 404s and the entire audit step fails.

This is a **chicken-and-egg problem**: the CI can't audit unreleased code because it needs the released binary, but the binary doesn't include the new code yet. The audit step is meant to run homeboy against the PR's changes, but it's running an old version of homeboy (or failing to download any version at all).

## Impact

- Every PR to homeboy itself gets a red Audit check that has nothing to do with the PR's code quality
- No inline review comments are posted (the step fails before running audit commands)
- The Build & Test job (cargo build + cargo test + cargo fmt) still works correctly since it compiles from source

## Possible fixes

1. **Build from source in CI** — instead of downloading a release binary, compile the PR branch. Slower but always correct for the homeboy repo itself.
2. **Fallback to cargo run** — if binary download fails, fall back to `cargo run -- audit` for the homeboy repo specifically.
3. **Skip audit on self** — detect when auditing the homeboy repo itself and use the just-built binary from the Build & Test step.
4. **Allow audit step to soft-fail on homeboy PRs** — mark the Audit check as non-required for the homeboy repo since Build & Test already validates compilation.

## Reproduction

Open any PR on `Extra-Chill/homeboy` — the Homeboy Audit job will fail with the 404.

## Logs

From PR #413, run `22676164467`:
```
Homeboy Audit  curl: (22) The requested URL returned error: 404
Homeboy Audit  ##[error]Process completed with exit code 22.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -43,16 +43,52 @@ jobs:
         run: cargo test
 
   # ── Homeboy audit (dogfooding) ──
-  # Compares audit findings against the committed baseline. Only fails
-  # if a PR introduces NEW drift (new findings not in the baseline).
-  # Resolving existing findings is always allowed (ratchet down).
+  # Build from source instead of downloading a release binary.
+  # This avoids the chicken-and-egg problem: homeboy-action downloads from
+  # the latest GitHub release, but that binary doesn't include the PR's code.
+  # For the homeboy repo itself, we compile the PR branch and audit with that.
+  # See: https://github.com/Extra-Chill/homeboy/issues/416
   audit:
     name: Homeboy Audit
+    needs: [build]
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - uses: Extra-Chill/homeboy-action@v1
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Cache cargo
+        uses: actions/cache@v4
         with:
-          extension: rust
-          commands: audit
-          component: homeboy
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+          restore-keys: ${{ runner.os }}-cargo-
+
+      - name: Build homeboy
+        run: cargo build
+
+      - name: Install extension and register component
+        run: |
+          BINARY=$(find target/debug -maxdepth 1 -name "homeboy" -type f | head -1)
+          if [ -z "$BINARY" ]; then
+            echo "::error::Could not find homeboy binary in target/debug"
+            exit 1
+          fi
+          chmod +x "$BINARY"
+
+          echo "Using $($BINARY --version)"
+
+          # Install the rust extension for audit rules
+          "$BINARY" extension install https://github.com/Extra-Chill/homeboy-extensions --id rust
+
+          # Register the component so audit can find it
+          "$BINARY" component create --local-path "$(pwd)" --extension rust
+
+      - name: Run audit
+        run: |
+          BINARY=$(find target/debug -maxdepth 1 -name "homeboy" -type f | head -1)
+          "$BINARY" audit homeboy

